sdks/go/pkg/beam/util/grpcx/hook.go

```diff
@@ -0,0 +1,88 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcx
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/util/hooks"
+	"google.golang.org/grpc"
+)
+
+// Hook allow a runner to customize various aspects of gRPC
+// communication with the FnAPI harness. Each member of the struct
+// is optional; the default behavior will be used if a value is not
+// supplied.
+type Hook struct {
+	// Dialer allows the runner to customize the gRPC dialing behavior.
+	Dialer func(context.Context, string, time.Duration) (*grpc.ClientConn, error)
+	// TODO(wcn): expose other hooks here.
+}
+
+type HookFactory func([]string) Hook
```

Comment:
<!--new_thread; commit:abd6cffe5451804ffdc2a1902dbdf7f3b588c57f; resolved:0-->
Exported types should have a comment.

// HookFactory configures a hook with the provided arguments.